### PR TITLE
sinkv2(cdc): seperate redo and sink memory quota

### LINKED ISSUE #8080: sink manager redo worker can be slow down by sink worker
### What did you do?

Create a changefeed with memory quota `2GB`, and run tpcc workload:
```bash
$ ./go-tpc tpcc --db test --warehouses 50 -T 20 --host main --port 4000 run
```

### What did you expect to see?

_No response_

### What did you see instead?

Redo write speed is not stable when memory usage is reach the given quota:
图片
图片

### Versions of the cluster

TiCDC master and release-6.5 is influenced.

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?

Issue Number: close #8080 

### What is changed and how it works?

#8080 happens because redo worker and sink worker share one memory quota. So if memory usage is close to the configured quota, redo can be blocked when acquiring memory.

This patch seperates redo and sink into 2 memory quota. So redo log worker can always get memory to write redo log.

With the patch, if user set a changefeed memory quota to `2GB`,
* redo memory quota will be 3/4, `1.5GB`;
* sink memory quota will be  1/4, `0.5GB`;
* redo cache will be 1/2 of redo memory quota, `0.75GB`;

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
    do same steps as #8080:
    图片 
    图片

### Release note 

```release-note
None
```

### CODE REVIEW

File: metrics/grafana/ticdc.json [hunk 1]
```diff
@@ -6485,9 +6485,9 @@
           "targets": [
             {
               "exemplar": true,
-              "expr": "ticdc_sinkmanager_memory_quota{k8s_cluster=\"$k8s_cluster\", tidb_cluster=\"$tidb_cluster\", changefeed=~\"$changefeed\", instance=~\"$ticdc_instance\"}",
+              "expr": "ticdc_sinkmanager_memory_quota{k8s_cluster=\"$k8s_cluster\", tidb_cluster=\"$tidb_cluster\", changefeed=~\"$changefeed\", component!=\"\", instance=~\"$ticdc_instance\"}",
```
Comment (amyangfei) [inline metrics/grafana/ticdc.json:6488]: Is `component!=\"\"` used for compatibility?

File: cdc/processor/sinkmanager/mem_quota.go [hunk 2]
```diff
@@ -53,14 +53,19 @@ type memQuota struct {
 	metricUsed  prometheus.Gauge
 }
 
-func newMemQuota(changefeedID model.ChangeFeedID, totalBytes uint64) *memQuota {
+func newMemQuota(changefeedID model.ChangeFeedID, totalBytes uint64, component ...string) *memQuota {
+	var comp string
+	if len(component) > 0 {
+		comp = component[0]
+	}
```
Comment (amyangfei) [inline cdc/processor/sinkmanager/mem_quota.go:60]: Can component be always passed in?

File: metrics/grafana/ticdc.json (hunk 1 above)
Comment (hicqu) [reply metrics/grafana/ticdc.json:6488]: Yes!

**Review by amyangfei — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**